### ISSUE TITLE
False positive: neg_cmp_op_on_partial_ord in macros

### ISSUE THREAD

**Issue by alexheretic** [rust-lang-nursery/rust-clippy#2856]
```rust
fn main() {
    let a = 1.0;
    let b = 2.0;

    assert!(b > a);
    assert!(a < b);
}
```

```
$ clippy
warning: The use of negated comparision operators on partially orded types produces code that is hard to read and refactor. Please consider to use the `partial_cmp` instead, to make it clear that the two values could be incomparable.
 --> src/main.rs:5:5
  |
5 |     assert!(b > a);
  |     ^^^^^^^^^^^^^^^
  |
  = note: #[warn(neg_cmp_op_on_partial_ord)] on by default

warning: The use of negated comparision operators on partially orded types produces code that is hard to read and refactor. Please consider to use the `partial_cmp` instead, to make it clear that the two values could be incomparable.
 --> src/main.rs:6:5
  |
6 |     assert!(a < b);
```

The lint is also lacking a help link to https://rust-lang-nursery.github.io/rust-clippy/master/index.html#neg_cmp_op_on_partial_ord.

**Comment by oli-obk** [comment]
cc @0ndorio 

The `cx.span_lint` should be clippy's `span_lint` helper.

Also we need a macro check on the negation. Or maybe we should just whitelist the `assert` macro?

**Comment by 0ndorio** [comment]
Thx for the cc, I am going to fix that.

> Also we need a macro check on the negation. Or maybe we should just whitelist the assert macro?

I am going to check if this triggers for all macros containing a related comparison. If so, I would favor to add a macro check instead of whitelist solution.
